util/src/main/java/tech/pegasys/artemis/util/SSZTypes/SSZList.java

```diff
@@ -13,48 +13,77 @@
 
 package tech.pegasys.artemis.util.SSZTypes;
 
-import com.fasterxml.jackson.databind.annotation.JsonSerialize;
-import java.util.ArrayList;
-import java.util.List;
+import com.google.common.base.Preconditions;
+import java.lang.reflect.Array;
+import java.util.Collection;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
-@JsonSerialize(as = ArrayList.class)
-public class SSZList<T> extends ArrayList<T> {
+public interface SSZList<T> extends SSZImmutableCollection<T> {
 
-  private long maxSize;
-  private Class<T> classInfo;
+  static <T> SSZMutableList<T> create(Class<? extends T> classInfo, long maxSize) {
```

Comment:
(optional) Perhaps we should make it more explicit that these methods are creating mutable lists:
```suggestion
  static <T> SSZMutableList<T> createMutable(Class<? extends T> classInfo, long maxSize) {
```

Reply:
👍